tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2083,3 +2083,43 @@ def count_below(x, t):
     :return type: float
     """
     return np.sum(x <= t)/len(x)
+
+
+@set_property("fctype", "simple")
+def benford_correlation(x):
+    """
+    Useful for anomaly detection applications [1][2]. Returns the correlation from first digit distribution when
+    compared to the Newcomb-Benford's Law distribution [3][4].
+
+    .. math::
+
+        P(d)=\\log_{10}\\left(1+\\frac{1}{d}\\right)
+
+    where :math:`P(d)` is the Newcomb-Benford distribution for :math:`d` that is the leading digit of the number {1, 2,
+    3, 4, 5, 6, 7, 8, 9}.
+
+    .. rubric:: References
+
+    |  [1] A Statistical Derivation of the Significant-Digit Law, Theodore P. Hill, Statistical Science, 1995
+    |  [2] The significant-digit phenomenon, Theodore P. Hill, The American Mathematical Monthly, 1995
+    |  [3] The law of anomalous numbers, Frank Benford, Proceedings of the American philosophical society, 1938
+    |  [4] Note on the frequency of use of the different digits in natural numbers, Simon Newcomb, American Journal of
+    |  mathematics, 1881
+
+   :param x: the time series to calculate the feature of
+   :type x: numpy.ndarray
+   :return: the value of this feature
+   :return type: float
+   """
+    if not isinstance(x, (np.ndarray, pd.Series)):
+        x = np.asarray(x)
+
+    x = np.trunc(x)  # retrieve first digit from data
+
+    # benford distribution
+    benford_distribution = np.array([0.301, 0.176, 0.125, 0.097, 0.079, 0.067, 0.058, 0.051, 0.046])
```

Comment:
Using "magical numbers" (= hard-coded number values) in code is not considered good practice for some reasons:
1. complicated to change, e.g. to add more significant digits (because I would need to re-calculate by hand)
2. error-prone (everyone reading your code would need to re-calculate to make sure, it is what the comment says)
3. we do not expect it here, but: if something in the original distribution changes, how can we make sure the correct values are used?

So I would propose the following:
* either calculate the distribution on the fly using its formula `log10(n + 1) - log10(n)`
* or calculate it once, store it into a constant outside of the function and use it here (preferred)